src/context/history-context.js

```diff
@@ -0,0 +1,73 @@
+import { createContext, useContext, useReducer } from 'react';
+import {
+    clearHistoryService,
+    getHistoryService,
+    removeFromHistoryService,
+    setHistoryService,
+} from '../services';
+import { useAuth } from './';
+import { historyReducer } from '../reducers';
+import toast from 'react-hot-toast'
+const HistoryContext = createContext();
+
+const HistoryProvider = ({ children }) => {
+    const AUTH_TOKEN = localStorage.getItem('TI_VID_AUTH_TOKEN');
+    const { auth } = useAuth();
+
+    const getHistoryHandler = async () => {
+        if (auth) {
+            const response = await getHistoryService(AUTH_TOKEN);
+            setHistory({ type: 'LOAD_HISTORY', payload: response.data.history });
+        }
+    };
+
+    const setHistoryHandler = async (video) => {
+        if (auth) {
+            const response = await setHistoryService(video, AUTH_TOKEN);
+            setHistory({ type: 'SET_HISTORY', payload: response.data.history });
+        }
+    };
+
+    const removeFromHistoryHandler = async (id) => {
+        if(auth){
+            const response = await removeFromHistoryService(id, AUTH_TOKEN);
+            setHistory({ type: 'SET_HISTORY', payload: response.data.history });
+            toast.success("Removed from watch history");
+        }
+    };
+
+    const clearHistoryHandler = async () => {
+        if(auth){
+            await clearHistoryService(AUTH_TOKEN);
+            setHistory({ type: 'CLEAR_HISTORY' });
+            toast.success("Cleared watch history");
+        }
+    };
+
+    const [history, setHistory] = useReducer(historyReducer, {
+        data: [],
+    });
+
+    return (
+        <HistoryContext.Provider
+            value={{
+                history,
+                setHistory,
+                getHistoryHandler,
```

Comment:
You can remove these 3 functions from the provider values if you're not using them in the context children.